regparser/citations.py

```diff
@@ -208,17 +209,24 @@ def single_citations(matches, comment):
         multiple_citations(
             grammar.multiple_section_paragraphs.scanString(text), False)
 
-    # Some appendix citations are... complex
-    for match, start, end in grammar.appendix_with_part.scanString(text):
-        full_start = start
-        if match.marker is not '':
-            start = match.marker.pos[1]
-        label = filter(lambda l: l != '.', list(match)[3:])
-        label = dict(zip(['p1', 'p2', 'p3'], label))
-        citations.append(ParagraphCitation(
-            start, end, initial_label.copy(
-                appendix=match.appendix, appendix_section=match.a1,
-                **label), full_start=full_start))
+    for gram in (grammar.appendix_with_part, grammar.appendix_par_of_part):
+        for match, start, end in gram.scanString(text):
+            full_start = start
+            if match.marker is not '':
+                start = match.marker.pos[1]
+            label = filter(lambda l: 1 != '.', list(match)[1:])
+            if match.appendix:
```

Comment:
If the logic here is broken into an `if` `else` due to the addition of `grammar.appendix_par_of_part`, why are they in the same `for` loop?